content/spin/running-apps.md

```diff
@@ -74,6 +75,45 @@ $ spin up --log-dir ~/dev/bugbash
+
+> The initial Spin `watch` command accepts valid Spin [up](https://developer.fermyon.com/common/cli-reference#up) options and passes them through to `spin up` for you, when running/rerunning the Spin application.
+
+By default, Spin watch monitors the application manifest (`spin.toml` file) and will rebuild the running application if the manifest changes. Spin watch also monitors any files specified in the `component.build.watch` section of the `spin.toml` file. For example, the following configuration (belonging to a Spin `http-rust` application) will be rebuilt via `cargo build --target wasm32-wasi --release` and then rerun using the initial `spin up` command whenever changes occur in either Rust (`.rs`) source files or the `Cargo.toml` file. Glob patterns (i.e. the use of the `*` wildcard) are accepted, as shown below:
+
+```toml
+[[component]]
+[component.build]
+command = "cargo build --target wasm32-wasi --release"
+watch = ["src/**/*.rs", "Cargo.toml"]
+```
+ If you would prefer Spin watch to only rerun the application (without a rebuild) when changes occur, you can use the `--skip-build` option when running the initial `spin watch` command.
+
+ Spin watch can will also monitor any changes to files specified in the `spin.toml`'s `component.files` areas and will rerun the application when changes occur. For example, any changes to `changing-file.txt` in the `my-files` directory will cause Spin watch to rerun the application:
+
+ ```toml
+[[component]]
+// -- snip
+files = ["my-files/changing-file.txt"]
+[component.build]
+command = "cargo build --target wasm32-wasi --release"
+watch = ["src/**/*.rs", "Cargo.toml"]
+```
+
+Spin watch will respond to filesystem events every 100 milliseconds, by default. This interval can be overridden by passing in the `--debounce` option i.e. `spin watch --debounce 1000` will make Spin watch respond to filesystem events once per second.
+
+> Note: Action is only taken after the `--debounce` time has elapsed. For example, when a watched file is changed using a `--debounce` value of 10000 (10 seconds) no effect will be seen in the application, until a full 10 seconds has elapsed; at which point rebuild and/or rerun may occur.
+
+> Note: If the build step (`spin build`) fails, `spin up` will not be run.
+
+Passing the `--clear` flag will clear the screen anytime a rebuild or rerun occurs. Spin watch will not clear the screen between rebuild and rerun as this provides you with an opportunity to see any warnings.
+
+> Note: Spin watch will always rebuild every component in your Spin application. It does not discern which individual component changed.
+
+Spin watch does not consider changes a file's metadata (when it was last modified) as a change
```

Comment:
Maybe move this sentence further up to where we are talking about files and reloads?  It feels a bit orphaned here on the other side of the `--clear` flag.  Or drop it altogether?  I'm not sure it's super important.

Reply:
I moved this up to where we talk about changes; more relevant.